### PR TITLE
Sprint 53.2

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 53.2

### Closed Issues 

- [6916 BUG: Add Paper Filing- Document type selected after initial save is not being saved ](https://github.com/flexion/ef-cms/issues/6916) 

### Open Issues 

- [6767 (NEEDED FOR FINAL) BUG: Document info not populated on Edit Docket Entry screen ](https://github.com/flexion/ef-cms/issues/6767) 
- [6857 BUG: Removed deadline still blocking case from trial ](https://github.com/flexion/ef-cms/issues/6857) 
- [6936 BUG: Notice of Docket Change : Language and format issues on generated PDF](https://github.com/flexion/ef-cms/issues/6936) 
- [6979 BUG: Deleting case note is also deleting judge's note ](https://github.com/flexion/ef-cms/issues/6979) 
- [7099 BUG: Document QC Inbox - clicking on document causing error ](https://github.com/flexion/ef-cms/issues/7099) 
- [7137 BUG: Blocked Report: Cases displaying in reports for all trial locations in the same state](https://github.com/flexion/ef-cms/issues/7137)

### CODE REVIEW

File: codeToggles.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+  7137: true,
 };
 
-const isCodeEnabled = issueNumber => {
-  return toggles[issueNumber];
-};
-
-module.exports.isCodeEnabled = isCodeEnabled;
+exports.isCodeEnabled = issueNumber => toggles[issueNumber];
+exports.isCodeDisabled = issueNumber => !toggles[issueNumber];
```
Comment (mmarcotte) [inline codeToggles.js:14]: mayhap simply `exports.isCodeDisabled = !isCodeEnabled;`

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            });
+
+          if (!caseRecord.trialSessionId) {
+            item.signedAt = applicationContext
+              .getUtilities()
+              .createISODateString();
+            item.signedJudgeName = CHIEF_JUDGE;
+            // Hard code signedByUserId as it's not really used for anything but
+            // is required when signedJudgeName has a value
+            item.signedByUserId = '60f2059d-3831-4ed1-b93b-8c8beec478a4';
```
Comment (mmarcotte) [inline web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js:44]: Is there a backlog item present for this conundrum? Do we need to look up the chiefJudge's userId?

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            item.signedByUserId = trialSession.judge.userId;
+          }
+
+          const updatedDocketEntry = new DocketEntry(item, {
+            applicationContext,
+          })
+            .validate()
+            .toRawObject();
+
+          itemsAfter.push({
```
Comment (mmarcotte) [inline web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js:72]: I think for any record that a migration runs on, it would be helpful to log its unique identifiers. Don't need to include it this time, but for future migrations it will be a requirement to be able to parse the logs and know how many records were touched as we could potentially predict that number beforehand.

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+
+    const results = await migrateItems([mockDocketEntry], documentClient);
+
+    expect(results).toMatchObject([
+      {
+        ...mockDocketEntry,
+        signedByUserId: MOCK_TRIAL.judge.userId,
+        signedJudgeName: MOCK_TRIAL.judge.name,
+      },
+    ]);
```
Comment (mmarcotte) [inline web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js:127]: Wouldn't we also expect signedAt to be a date value?

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+      eventCode:
+        SYSTEM_GENERATED_DOCUMENT_TYPES.standingPretrialOrder.eventCode,
+      isMinuteEntry: undefined,
+      signedAt: '2020-07-06T17:06:04.552Z',
+    };
+
+    await expect(
+      migrateItems([mockDocketEntry], documentClient),
+    ).rejects.toThrow(
+      'The DocketEntry entity was invalid. {"signedByUserId":"\'signedByUserId\' must be a valid GUID"}',
```
Comment (mmarcotte) [inline web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js:245]: Hm, even though the problem is the Trial Session's fault... it's hard to figure that out by the error thrown. For migrations like this in the future, it would be helpful to have that sort of situation spelled out. If the migration fails, we will have to write another migration to remediate the problems (in this case, it would be to lookup the judgeId by their name?) Guess this goes back to the earlier question about whether or not there's a backlog item about making proper use of signedByUserId.

File: web-api/terraform/bin/deploy-app.sh [hunk 6]
```diff
@@ -51,13 +51,14 @@ fi
 
 npm run build:assets
 
+# exit on any failure
+set -eo pipefail
```
Comment (mmarcotte) [inline web-api/terraform/bin/deploy-app.sh:55]: thanks for these! 😄

**Review by mmarcotte — COMMENTED** [review]
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [ ] If changes were made in the UI (if not, skip):
  - [ ] If desired, run `docker system prune` to remove any unused docker images from previous code review.
  - [ ] Fetch the pull request for the sprint (e.g., `git fetch origin pull/{PR #}/head:sprint-{Sprint #}`), and then switch to that branch (e.g. `git checkout sprint-{Sprint #}`)
  - [ ] stand up the site locally, with `./docker-run.sh`
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
	- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
	- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/)
	- [ ] test in Mobile Safari and Mobile Chrome (or an emulator like Chrome DevTools), with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
	- [ ] user-visible changes (including API users like the IRS) are documented in CHANGELOG.md (which is linked from US Tax Court website).
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines

- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: web-client/src/views/viewFactory.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  Button,
+  CaseNotes,
+  ErrorNotification,
+};
+
+if (isCodeDisabled(6979)) {
+  views.CaseNotes = CaseNotesOld;
+}
+
+export const getView = viewName => views[viewName];
```
Comment (mmarcotte) [inline web-client/src/views/viewFactory.js:20]: Cool stuff. Will keep an eye on these factories. 👍 

factory

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js (hunk 2 above)
Comment (rachaelparris) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js:44]: We don't use this id for anything, so we didn't think it was necessary. If you disagree we can definitely make that change. This will really only affect old data on dev, though.

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js (hunk 4 above)
Comment (rachaelparris) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js:127]: `signedAt` was already set on the docket entry, so we expect it to be the same

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js (hunk 5 above)
Comment (rachaelparris) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js:245]: I'm not sure how we could know that the trial session was the issue here and throw a different error than just our normal validation error. Should we validate the trial session that comes back from persistence first? This test was meant to verify that the docket entry is validated before it's sent to persistence, so we just needed to make something on the docket entry invalid. It didn't necessarily have to be an issue with the trial session that caused the validation error.

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js (hunk 2 above)
Comment (mmarcotte) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js:44]: Right on. I don't think it needs to change in this PR or in this migration script. I just want to be sure we don't make assumptions about `signedByUserId`, as this comment and update suggests that it's basically required junk data.  And so wanting to be sure there's a backlog item to make it work for us (or get removed) in the future.

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js (hunk 4 above)
Comment (mmarcotte) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js:127]: Right on, but looking at the logic of the migration script, the timestamp would in fact change.

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js (hunk 2 above)
Comment (rachaelparris) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js:44]: That is a good point! I will backlog looking into the requirement for `signedByUserId` and if it is even necessary.

Comment (rachaelparris) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js:44]: I looked some more, and `signedByUserId` is probably required because it is the user id of the _user_ who signed the document, not the judge. So I think it is probably useful when an actual user signs a document to know who signed it. But it will just be junk data when coming from a migration, because the migration is the one signing it. 😄

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js (hunk 4 above)
Comment (rachaelparris) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js:127]: Aha, you are correct! Would you like for us to add that expectation?

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js (hunk 2 above)
Comment (mmarcotte) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.js:44]: Thanks!

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js (hunk 4 above)
Comment (mmarcotte) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js:127]: No, it's okay in this case as we will hopefully not be having to run this migration. Thank you, though! Helpful for me to walk through these with you.

File: web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js (hunk 5 above)
Comment (mmarcotte) [reply web-api/migration-terraform/main/lambdas/migrations/0004-standing-pretrial-order-signatures.test.js:245]: I think this is good enough for now. I'm going to have to be more vigilant with understanding exactly what's happening these migrations going forward as they will be affecting live data. Thanks for taking a look!

**Review by mmarcotte — APPROVED** [review]
